topic_statistics_demo/src/imu_talker_listener_nodes.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <cstdlib>
+#include <string>
+
+#include "topic_statistics_demo/imu_talker_listener_nodes.hpp"
+
+using namespace std::chrono_literals;
+
+ImuTalker::ImuTalker(
+  const std::string & topic_name,
+  std::chrono::milliseconds publish_period)
+: Node("imu_talker"),
+  topic_name_(topic_name),
+  publish_period_(publish_period),
+  random_number_seed_(time(0)) {}
+
+void ImuTalker::initialize()
+{
+  RCLCPP_INFO(get_logger(), "Talker starting up");
+
+  srand(time(0));
+  publisher_ = this->create_publisher<sensor_msgs::msg::Imu>(
+    topic_name_,
+    10 /* QoS history_depth */);
+  publish_timer_ = create_wall_timer(
+    publish_period_,
+    [this]() -> void {
+      publish();
+    });
+}
+
+void ImuTalker::publish()
+{
+  sensor_msgs::msg::Imu msg;
+  // Timestamp the message to a random time before now,
```

Comment:
Is this indicative of what someone would see on a system? You could subtract a set percentage of the publish period only x% of the time, so you're introducing spurious samples that affects the deviation the most. This percentage could vary from 0-100% for each publish period (uniform distribution perhaps?).

IMHO this better reflects the delay of a message received by a subscriber, especially if you log the delay that you're introducing to the user.

Reply:
Added random distribution and fix percentage difference.